packages/core/src/actions/execute.ts

```diff
@@ -321,24 +475,90 @@ export const approveDeploymentViaSigner: ApproveDeployment = async (
   approvalLeafWithProof,
   executionMode,
   provider,
-  signer
+  ownerSignatures,
+  deploymentContext
 ) => {
-  const ownerSignature = await signMerkleRoot(merkleRoot, signer)
+  const sphinxModule = new ethers.Contract(moduleAddress, SphinxModuleABI)
 
-  const sphinxModule = new ethers.Contract(
-    moduleAddress,
-    SphinxModuleABI,
-    signer
+  const packedOwnerSignatures = ethers.solidityPacked(
+    new Array(ownerSignatures.length).fill('bytes'),
+    ownerSignatures
+  )
+  const approvalData = sphinxModule.interface.encodeFunctionData('approve', [
+    merkleRoot,
+    approvalLeafWithProof,
+    packedOwnerSignatures,
+  ])
+
+  return deploymentContext.executeTransaction(
+    deploymentContext,
+    {
+      to: moduleAddress,
+      data: approvalData,
+      chainId: (await provider.getNetwork()).chainId.toString(),
+    },
+    executionMode
   )
+}
 
-  return (
-    await sphinxModule.approve(
-      merkleRoot,
-      approvalLeafWithProof,
-      ownerSignature,
-      await getGasPriceOverrides(provider, signer, executionMode)
+export const executeTransactionViaSigner: ExecuteTransaction = async (
+  deploymentContext: DeploymentContext,
+  transaction: MinimumTransaction,
+  executionMode: ExecutionMode
+): Promise<ethers.TransactionReceipt> => {
+  const { wallet } = deploymentContext
+
+  if (!wallet) {
+    throw new Error(
+      'No signer passed to executeTransaction. This is a bug, please report it to the developers.'
+    )
+  }
+
+  const txReceipt = await (
+    await wallet.sendTransaction(
+      await getGasPriceOverrides(
+        deploymentContext.provider,
+        wallet,
+        executionMode,
+        transaction
+      )
     )
   ).wait()
+
+  if (txReceipt === null) {
+    throw new Error(
+      'No transaction receipt returned by ethers. This is a bug, please report it to the developers.'
+    )
+  }
+
+  return txReceipt
+}
+
+export const injectRoles: InjectRoles = async (
```

Comment:
Used to set the relayer role and gnosis safe owners in the simulation and when deploying on a local node.